generator/internal/language/rusttemplate.go

```diff
@@ -149,6 +148,23 @@ type RustEnumValue struct {
 	EnumType string
 }
 
+type RustCodecVisitor struct {
```

Comment:
Ah, here is your second visitor... We are getting closer to having the abstraction pay off [^1].


[^1]: my rule of thumb (stolen from a book I think) is that you need 3 non-trivial use-cases to break even, and 4 to actually gain something.

Reply:
I think we could use this pattern for all language templates going forward. It reduces duplication, as you don't have to navigate to the elements you want to decorate, just override the method for the element you want. 
I didn't want to go all-in in this PR, but I think we could remove some code from the `rusttemplate.go` and `gotemplate.go` using this.

Reply:
I'll resolve this to keep the conversation in a single place